src/main/scala/com/advancedtelematic/director/http/AdminResource.scala

```diff
@@ -260,21 +269,31 @@ class AdminResource(extractNamespace: Directive1[Namespace], val keyserverClient
       } ~
       path("multi_target_update" / UpdateId.Path) { updateId =>
         put {
-          setMultiUpdateTarget(ns, device, updateId)
+          correlationIdParameter(updateId) { correlationId =>
+            setMultiUpdateTarget(ns, device, updateId, correlationId)
+          }
         }
       }
     }
 
   def multiTargetUpdatesRoute(ns: Namespace): Route =
     pathPrefix("multi_target_updates" / UpdateId.Path) { updateId =>
-      (pathEnd & get) {
-        getMultiTargetUpdates(ns, updateId)
-      } ~
-      (pathEnd & put & entity(as[Seq[DeviceId]])) { devices =>
-        setMultiTargetUpdateForDevices(ns, devices, updateId)
+      pathEnd {
+        get {
+          getMultiTargetUpdates(ns, updateId)
+        } ~
+        put {
+          entity(as[Seq[DeviceId]]) { devices =>
+            correlationIdParameter(updateId) { correlationId =>
```

Comment:
when do we ever need to pass a correlation id here? is kind of weird to pass params in the url + path + body.

Reply:
Good point. Deprecating these old endpoints. Adding new endpoint `update_devices` that take parameters in PUT request body.